source/Server/DirectoryServices/UserPrincipalWrapper.cs

```diff
@@ -1,14 +1,14 @@
 using System;
 using System.Collections.Generic;
 using System.DirectoryServices.AccountManagement;
-using System.Linq;
+using System.Threading;
 
 namespace Octopus.Server.Extensibility.Authentication.DirectoryServices.DirectoryServices
 {
     interface IUserPrincipalWrapper : IPrincipalWrapper, IDisposable
     {
-        IEnumerable<IPrincipalWrapper> GetAuthorizationGroups();
-        IEnumerable<IPrincipalWrapper> GetGroups();
+        IList<IPrincipalWrapper> GetAuthorizationGroups(CancellationToken cancellationToken);
```

Comment:
Should we not still be using `IEnumerable` here?

Reply:
I don't see a reason to use IEnumerable

Reply:
Just for the more general interface - we aren't using any features of IList.